### PR TITLE
Xfail tests for ndarray indexing under HIP

### PR DESCRIPTION
**Pull request by takagi**
Rel #4132.

### CODE REVIEW

File: tests/cupy_tests/core_tests/test_ndarray_indexing.py [hunk 1]
```diff
@@ -78,9 +78,27 @@
 @testing.gpu
 class TestArrayIndexingParameterized(unittest.TestCase):
 
+    _getitem_hip_skip_condition = [
+        ((1, 0, 2), (2, 3, 4), None),
+        ((-1, 0, -2), (2, 3, 4), None),
+        ((1, 0, 2), (2, 3, 4), (2, 0, 1)),
+        ((-1, 0, -2), (2, 3, 4), (2, 0, 1)),
+        ((slice(None, None, None), None), (2,), None),
+        ((slice(-9, -10, -1),), (10,), None),
```
Comment (leofang) [inline tests/cupy_tests/core_tests/test_ndarray_indexing.py:87]: Is this entry new? I haven't seen that earlier (https://github.com/cupy/cupy/issues/4484#issuecomment-773458951).

**Comment by leofang** [comment]
Fix part of https://github.com/cupy/cupy/issues/4484#issuecomment-773458951.

File: tests/cupy_tests/core_tests/test_ndarray_indexing.py (hunk 1 above)
Comment (takagi) [reply tests/cupy_tests/core_tests/test_ndarray_indexing.py:87]: Yes. It newly fails when I make other entries skip. It's strange as the test should not change any state. Would you check if you can reproduce it if possible?

**Comment by leofang** [comment]
This is *super* strange: On ROCm 3.5.0 + Radeon VII, all tests in `test_ndarray_indexing.py` passed without any problem, but on ROCm 4.0.0 + MI50 I saw those failures.

File: tests/cupy_tests/core_tests/test_ndarray_indexing.py (hunk 1 above)
Comment (leofang) [reply tests/cupy_tests/core_tests/test_ndarray_indexing.py:87]: I can reproduce this on ROCm 4.0.0 + MI50 (see my other comment).

**Comment by leofang** [comment]
> This is _super_ strange: On ROCm 3.5.0 + Radeon VII, all tests in `test_ndarray_indexing.py` passed without any problem, but on ROCm 4.0.0 + MI50 I saw those failures.

All tests in `test_ndarray_indexing.py` passed on ROCm 4.0.0 + Radeon VII Pro.

File: tests/cupy_tests/core_tests/test_ndarray_indexing.py (hunk 1 above)
Comment (leofang) [reply tests/cupy_tests/core_tests/test_ndarray_indexing.py:87]: On another MI50 machine (not the one we both have access to), I am able to see all 7 tests failing (including this `(-9, -10, -1)` one).

**Comment by leofang** [comment]
Saw more failures on MI100...@takagi could you try to reproduce?
```
======================================================================= short test summary info =======================================================================
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_0_{indexes=(1, 0, 2), shape=(2, 3, 4), transpose=None}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_1_{indexes=(-1, 0, -2), shape=(2, 3, 4), transpose=None}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_2_{indexes=(1, 0, 2), shape=(2, 3, 4), transpose=(2, 0, 1)}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_3_{indexes=(-1, 0, -2), shape=(2, 3, 4), transpose=(2, 0, 1)}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_10_{indexes=(1, Ellipsis), shape=(2, 3, 4), transpose=None}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_13_{indexes=(None,), shape=(2, 3), transpose=None}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_14_{indexes=(slice(None, None, None), None), shape=(2,), transpose=None}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_18_{indexes=(slice(-9, -10, -1),), shape=(10,), transpose=None}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_27_{indexes=(slice(-4, -5, -1),), shape=(10,), transpose=None}::test_getitem
FAILED tests/cupy_tests/core_tests/test_ndarray_indexing.py::TestArrayIndexingParameterized_param_38_{indexes=(slice(-5, -6, -1),), shape=(10,), transpose=None}::test_getitem
============================================================== 10 failed, 58 passed in 65.11s (0:01:05) ===============================================================
```

**Comment by leofang** [comment]
I feel something is wrong on these MI*** cards...

**Comment by takagi** [comment]
> Saw more failures on MI100...@takagi could you try to reproduce?

It is more than those I can see on Vega 20...

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
pfnCI, test this please.

**Pull request merged**